buildpipeline/perf-pipeline.groovy

```diff
@@ -7,188 +7,200 @@
 //--------------------- Windows Functions ----------------------------//
 
 def windowsBuild(String arch, String config, String pgo, boolean isBaseline) {
-    checkout scm
+    simpleNode('Windows_NT','latest') {
+        checkout scm
 
-    String pgoBuildFlag = ((pgo == 'nopgo') ? '-nopgooptimize' : '-enforcepgo')
-    String baselineString = ""
+        String pgoBuildFlag = ((pgo == 'nopgo') ? '-nopgooptimize' : '-enforcepgo')
+        String baselineString = ""
 
-    // For baseline builds, checkout the merge's parent
-    if (isBaseline) {
-        baselineString = "-baseline"
-        bat "git checkout HEAD^^1"
-    }
+        // For baseline builds, checkout the merge's parent
+        if (isBaseline) {
+            baselineString = "-baseline"
+            bat "git checkout HEAD^^1"
+        }
 
-    bat "set __TestIntermediateDir=int&&.\\build.cmd -${config} -${arch} -skipbuildpackages ${pgoBuildFlag}"
+        bat "set __TestIntermediateDir=int&&.\\build.cmd -${config} -${arch} -skipbuildpackages ${pgoBuildFlag}"
 
-    // Stash build artifacts. Stash tests in an additional stash to be used by Linux test runs
-    stash name: "nt-${arch}-${pgo}${baselineString}-build-artifacts", includes: 'bin/**'
-    stash name: "nt-${arch}-${pgo}${baselineString}-test-artifacts", includes: 'bin/tests/**'
+        // Stash build artifacts. Stash tests in an additional stash to be used by Linux test runs
+        stash name: "nt-${arch}-${pgo}${baselineString}-build-artifacts", includes: 'bin/**'
+        stash name: "nt-${arch}-${pgo}${baselineString}-test-artifacts", includes: 'bin/tests/**'
+    }
 }
 
 def windowsPerf(String arch, String config, String uploadString, String runType, String opt_level, String jit, String pgo, String scenario, boolean isBaseline) {
-    withCredentials([string(credentialsId: 'CoreCLR Perf BenchView Sas', variable: 'BV_UPLOAD_SAS_TOKEN')]) {
-        checkout scm
-        String baselineString = ""
-        if (isBaseline) {
-            baselineString = "-baseline"
-        }
-        dir ('.') {
-            unstash "nt-${arch}-${pgo}${baselineString}-build-artifacts"
-            unstash "benchview-tools"
-            unstash "metadata"
-        }
+    simpleNode('windows_server_2016_clr_perf', 180) {
```

Comment:
Note: This would mean ILLink jobs would go on the perf machines. Not sure if we actually want that or not.

Reply:
I am updating the label on https://github.com/dotnet/coreclr/pull/14900. This was illink will run on VMs.

Reply:
Right, but I've moved the simpleNode lines out of the individual calls to windowsPerf, etc. into windowsPerf, so if we want a different node type for illink I'll have to do some experimenting/changing of how we do this part.

Reply:
We definitely do not want to use real hardware where it is not needed.

Reply:
Ok, I'll try to finagle something so that we can use something different for illink.